app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  # Method to update the posts counter for a user.
+  def update_posts_counter
+    user.update(posts_counter: user.posts.count)
+  end
```

Comment:
- [ ] [optional] @edubew great work adding the update_posts_counter method, you could also consider using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_posts_counter method every time a new post has been saved to the database, then make the update_posts_counter method private, this applies to like and comment counters also.

```
 after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```